From b966264fb615a46a6e57e8b6d5d9308f5c383250 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Wed, 17 Jan 2024 21:11:16 +0100
Subject: [PATCH 07/26] drm/bridge: imx: add resume support for i.MX8MP DWC
 HDMI

Add missing dw_hdmi_resume() call to the wrapper resume path,
without this the analog part of the HDMI PHY is not ungated
and the PHY PLL would not relock on resume. This will be
squashed into the next posting of wrapper driver.

Upstream-Status: Pending [Will be squashed into "drm/bridge: imx: add bridge wrapper driver for i.MX8MP DWC HDMI"]
Signed-off-by: Marek Vasut <marex@denx.de>
---
 drivers/gpu/drm/bridge/imx/imx8mp-hdmi.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/drivers/gpu/drm/bridge/imx/imx8mp-hdmi.c b/drivers/gpu/drm/bridge/imx/imx8mp-hdmi.c
index 59ea67d502180..a19ad9cb4b894 100644
--- a/drivers/gpu/drm/bridge/imx/imx8mp-hdmi.c
+++ b/drivers/gpu/drm/bridge/imx/imx8mp-hdmi.c
@@ -118,6 +118,25 @@ static int imx8mp_dw_hdmi_remove(struct platform_device *pdev)
 	return 0;
 }
 
+static int __maybe_unused imx8mp_dw_hdmi_pm_suspend(struct device *dev)
+{
+	return 0;
+}
+
+static int __maybe_unused imx8mp_dw_hdmi_pm_resume(struct device *dev)
+{
+	struct imx8mp_hdmi *hdmi = dev_get_drvdata(dev);
+
+	dw_hdmi_resume(hdmi->dw_hdmi);
+
+	return 0;
+}
+
+static const struct dev_pm_ops imx8mp_dw_hdmi_pm_ops = {
+	SET_SYSTEM_SLEEP_PM_OPS(imx8mp_dw_hdmi_pm_suspend,
+				imx8mp_dw_hdmi_pm_resume)
+};
+
 static const struct of_device_id imx8mp_dw_hdmi_of_table[] = {
 	{ .compatible = "fsl,imx8mp-hdmi" },
 	{ /* Sentinel */ },
@@ -130,6 +149,7 @@ static struct platform_driver imx8mp_dw_hdmi_platform_driver = {
 	.driver		= {
 		.name	= "imx8mp-dw-hdmi",
 		.of_match_table = imx8mp_dw_hdmi_of_table,
+		.pm = &imx8mp_dw_hdmi_pm_ops,
 	},
 };
 
-- 
2.43.0

